pkg/config/agent/agent.go

```diff
@@ -84,7 +84,16 @@ func DefaultConfig() Config {
 // config struct with merged configuration from all provided files.
 func ParseConfigFiles(files []string) (Config, error) {
 	c := DefaultConfig()
-	return c, cfgutil.ParseYAML(&c, files...)
+	err := cfgutil.ParseYAML(&c, files...)
+
+	// If the config YAML has a blank S3 key with nothing defined it will still
+	// overwrite the default config, and not use the IAM role. Fall back to the
+	// default if that is the case.
+	if (rclone.S3Options{}) == c.S3 {
```

Comment:
I guess this is true for all options

Reply:
Yeah - if any of these exist in the YAML with no properties it will overwrite everything.

From what I can tell the other providers don't have an option to use authentication details from the environment though so it wouldn't have the same effect of breaking something that works by default.